### ISSUE TITLE
Investigate why we don't see any catastrophic forgetting 

### ISSUE THREAD

**Issue by RomainGoussault** [SubstraFoundation/distributed-learning-contributivity#140]
We should have come across [Catastrophic forgetting](https://ai.stackexchange.com/a/13293
) when doing experiment with the [seq ](https://github.com/SubstraFoundation/distributed-learning-contributivity#configuration-of-the-collaborative-and-distributed-learning)approach.

We did not, and I think we should aim to provoke this Catastrophic forgetting, to be sure our distributed learning code is not flawed. I think we should:

- Have a real sequential approach (without any averaging)
- Run simulation on stratified datasets with different minibatch sizes.

cc @bowni @Thomas-Galtier

**Comment by bowni** [comment]
I tried a few scenarios, indeed it seems difficult to observe. An hypothesis: because the validation dataset used in the `fit()` is global and representative of the whole data space; with a 'local' validation dataset corresponding to the data space of the local partner, we would overfit it at each round and observe the 'forgetting'.

What do you think @RomainGoussault @Thomas-Galtier ?

**Comment by RomainGoussault** [comment]
Below some results with the `seq-pure` approach:
![image](https://user-images.githubusercontent.com/5354301/83856863-7c065700-a71a-11ea-9f7a-b7adb4e6badc.png)

The trends looks OK but I am unsure if we should see such good results with the pure-seq approch with one minibatch.


Two things we can try to investigate further:

- Look at the loss for the pure-seq approch with one minibatch.
- Implement a new file that learns on MNIST (without any multi partner learning) but with a stratified dataste and check if we see any castastrophic forgetting.

**Comment by RomainGoussault** [comment]
Also we should try with a more stratified scenario with 10 partner, each one with only one digit.

**Comment by RomainGoussault** [comment]
We don't see much in the loss --> https://github.com/SubstraFoundation/distributed-learning-contributivity/blob/486f58b91fd946ac1cf766fa1a859a9a3adc9796/notebooks/mpl_with_losses.ipynb

**Comment by RomainGoussault** [comment]
I launched the following new experiment:
```
   amounts_per_partner: 
     - [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
   samples_split_option: 
     - [[1, 'specific'], [1, 'specific'], [1, 'specific'], [1, 'specific'], [1, 'specific'], [1, 'specific'], [1, 'specific'], [1, 'specific'], [1, 'specific'], [1, 'specific']]
```

**Comment by RomainGoussault** [comment]
Results:
![image](https://user-images.githubusercontent.com/5354301/84010110-7e63ed80-a974-11ea-82e9-e6cf150928c9.png)

**Comment by RomainGoussault** [comment]
No, `val_data `is only used at the end of each epoch to evaluate the model:

From https://keras.io/api/models/model_training_apis/#fit-method:
`validation_data: Data on which to evaluate the loss and any model metrics at the end of each epoch. The model will not be trained on this data.`

**Comment by bowni** [comment]
@RomainGoussault could you put it in the saved experiments and name it Catastrophic Forgetting?
